include/dd/Operations.hpp

```diff
@@ -107,137 +107,57 @@ getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
     std::swap(target0, target1);
   }
 
-  if (controls.empty()) {
-    // the DD creation without controls is faster, so we use it if possible
-    // and only use the DD creation with controls if necessary
-    TwoQubitGateMatrix gm;
-    bool definitionFound = true;
-    switch (type) {
-    case qc::SWAP:
-      gm = SWAP_MAT;
-      break;
-    case qc::iSWAP:
-      gm = inverse ? ISWAPDG_MAT : ISWAP_MAT;
-      break;
-    case qc::iSWAPdg:
-      gm = inverse ? ISWAP_MAT : ISWAPDG_MAT;
-      break;
-    case qc::DCX:
-      gm = DCX_MAT;
-      break;
-    case qc::ECR:
-      gm = ECR_MAT;
-      break;
-    case qc::RXX:
-      gm = inverse ? rxxMat(-parameter[0U]) : rxxMat(parameter[0U]);
-      break;
-    case qc::RYY:
-      gm = inverse ? ryyMat(-parameter[0U]) : ryyMat(parameter[0U]);
-      break;
-    case qc::RZZ:
-      gm = inverse ? rzzMat(-parameter[0U]) : rzzMat(parameter[0U]);
-      break;
-    case qc::RZX:
-      gm = inverse ? rzxMat(-parameter[0U]) : rzxMat(parameter[0U]);
-      break;
-    case qc::XXminusYY:
-      gm = inverse ? xxMinusYYMat(-parameter[0U], parameter[1U])
-                   : xxMinusYYMat(parameter[0U], parameter[1U]);
-      break;
-    case qc::XXplusYY:
-      gm = inverse ? xxPlusYYMat(-parameter[0U], parameter[1U])
-                   : xxPlusYYMat(parameter[0U], parameter[1U]);
-      break;
-    default:
-      definitionFound = false;
-    }
-    if (definitionFound) {
-      return dd.makeTwoQubitGateDD(gm, nqubits, target0, target1, startQubit);
-    }
-  }
-
+  TwoQubitGateMatrix gm;
   switch (type) {
   case qc::SWAP:
-    // SWAP is self-inverse
-    return dd.makeSWAPDD(nqubits, controls, target0, target1, startQubit);
+    gm = SWAP_MAT;
+    break;
   case qc::iSWAP:
-    if (inverse) {
-      return dd.makeiSWAPinvDD(nqubits, controls, target0, target1, startQubit);
-    }
-    return dd.makeiSWAPDD(nqubits, controls, target0, target1, startQubit);
```

Comment:
The corresponding routines in the `Package.hpp` file can all be deleted now. That's actually what I meant with one of my earlier comments that all these functions for creating controlled two-target functions can be extracted into tests.
Just remove all that code from the main package (`Package.hpp`) and move it to the tests. You can just copy (and slighlty adjust) the definitions from the `Package.hpp` file and move them to the `test_dd_functionality.cpp` test file.
This should reduce the number of functions in the main package file considerably.

Reply:
Done 👍🏼